docker/docker_image_src.go

```diff
@@ -129,9 +129,42 @@ func (s *dockerImageSource) ensureManifestIsLoaded() error {
 	return nil
 }
 
+func (s *dockerImageSource) getExternalBlob(urls []string) (io.ReadCloser, int64, error) {
+	var (
+		resp *http.Response
+		err  error
+	)
+	for _, url := range urls {
+		resp, err = s.c.makeRequestToResolvedURL("GET", url, nil, nil, -1)
```

Comment:
I know I asked about authentication; still, is this right?

This would, when pulling a foreign RHEL from `docker.io`, send the `docker.io` credentials to Red Hat.